packages/vite/src/modes/global/dev.ts

```diff
@@ -146,7 +148,14 @@ export function GlobalModeDevPlugin({ uno, tokens, tasks, flushTasks, affectedMo
           tasks.push(extract(code, filename))
         },
       },
-      resolveId(id) {
+      resolveId(id, importer) {
+        if (
+          isAstro
+          && importer === 'astro:scripts/page-ssr.js'
+          && id.includes('?uno')
+        )
+          return `${id}?uno`
```

Comment:
Emm. I don't think this is a good approach to have Astro-specific logic inside the Vite plugin.

Reply:
I think so too, but I haven't thought of a better way.